### DISCUSSION TITLE
Detail page for locations should display card clicked on

### DISCUSSION THREAD

**Discussion by aaaleeenaaa** [neuefische/questions#586] [Web] [answered]
Hello everybody, 

I am currently working on [this US](https://github.com/aaaleeenaaa/capstone-project-bee-organized/issues/3). Clicking on a location should lead to a detail page where I first want to show the respective card I clicked on. This should be a heading for the list of bee colonies that will follow later on. 

Currently, clicking on locations leads to a detail page and on this page, a card is rendered but unfortunately it is empty. It should contain the name of the location clicked on. 

As adding all the relevant code here would be too much, I created a [pull request](https://github.com/aaaleeenaaa/capstone-project-bee-organized/pull/11) where you can see the current status.

(On a side note: The vercel deployment for the pull request shows an error (Command "npm run build" exited with 1) and in the details, it says that "location" wouldn't be defined. Maybe that is a first hint on what the problem is?)

Thanks in advance!

**Comment by NikRoe** [comment]
Hi Alena,

you are very close and one of the issues is indeed related to the error message you receive from vercel. 
In your dynamic route "locataiondetail/[id].js" you are currently trying to pass a prop that does not exist (location).

It needs to be the result that you pass:
```jsx
<Card location={result} />
```

However, this won't fix the whole issue yet. You are trying to filter based on an id.
Your locations are currently only strings.

If you want to filter based on the id, you need to update your locations to be objects instead of strings. They could look like this
```js
{
id: "cjklösdfieopsirop",
location: "Hamburg",
}
```
you would need to update your handleAddLocation() function to not just add a string, but an object to the state.

like this:
```js
setLocations([...locations, {id: nanoid(), location: newLocation}]);
```

nanoid is a library you can use to generate ids, you need to install it via npm prior to using it.

Hope this helps =)

**Reply by aaaleeenaaa** [reply, marked as answer]
Hello Niklas, thanks a lot!

Just a minute before I saw your answer, I was able to fix it with the help of one of the group members.

He suggested 

``
const result = locations[id];
``

for the location detail page. This then needs to be passed to the return statement as well ( `<Card location={result} />` ) and the card component needed to be fixed like that: 


```
export default function Card({ location, id }) {

  return (
    <StyledListElement>
      <CardElement id={id}>
        <p>{location.location}</p>
      </CardElement>
    </StyledListElement>
  );
}

```


But I will definitely add the nanoid, that's a very good point. Thank you so much for your help!
